### DISCUSSION TITLE
Pixelated Edges in Rectangle when rotating camera.

### DISCUSSION THREAD

**Discussion by Vipul-Cariappa** [raysan5/raylib#1763] [Q&A] [answered]
When I am rotating the camera and drawing rectangle, the edges of the rectangle is not smooth and is pixelated. Is there a anti aliasing method to improve the smoothness of the rectangles. If my question is not clear please forgive me as I am a beginner to c programing language and game development.

When zoomed into the image you can see it being pixelated but when running it look very bad when compared to this image:

![Screenshot (5)](https://user-images.githubusercontent.com/55694469/117668769-e265e600-b1c3-11eb-8aa1-a0f98c4a30a2.png)

part of source code:
```
// initializing camera
Camera2D camera = { 0 };
camera.target = (Vector2) {player.x, player.y};
camera.offset = (Vector2){ screenWidth/2.0f, screenHeight/2.0f };
camera.rotation = 0.0f;
camera.zoom = 1.0f;

// inside game loop
camera.rotation += delta*2;
```
Visit [Game](https://vipul-cariappa-production.up.railway.app/game/)(extream Mode) to see it.

**Comment by raysan5** [comment, marked as answer]
You can enable MSAA calling `SetConfigFlags(FLAG_MSAA_4X_HINT)` before `InitWindow()`.

Another option is using an FXAA shader.

**Reply by Vipul-Cariappa** [reply]
Thank You for the reply. MSAA works fine.
